Moonscraper Chart Editor/Assets/Scripts/Game/Charts/IO/Midi/MidReader.cs

```diff
@@ -926,52 +965,89 @@ static void ProcessNoteOnEventAsForcedTypePostDelay(in EventProcessParams eventP
             SongObjectHelper.GetRange(chart.notes, tick, tick + endPos, out index, out length);
 
             uint lastChordTick = uint.MaxValue;
+            bool expectedForceFailure = true; // Whether or not it is expected that the actual type will not match the expected type
             bool shouldBeForced = false;
 
             for (int i = index; i < index + length; ++i)
             {
+                // Tap marking overrides all other forcing
                 if ((chart.notes[i].flags & Note.Flags.Tap) != 0)
                     continue;
 
                 Note note = chart.notes[i];
 
+                // Check if the chord has changed
                 if (lastChordTick != note.tick)
                 {
+                    expectedForceFailure = false;
                     shouldBeForced = false;
 
-                    if (noteType == Note.NoteType.Strum)
+                    switch (noteType)
                     {
-                        if (!note.isChord && note.isNaturalHopo)
+                        case (Note.NoteType.Strum):
                         {
-                            shouldBeForced = true;
+                            if (!note.isChord && note.isNaturalHopo)
+                            {
+                                shouldBeForced = true;
+                            }
+                            break;
                         }
-                    }
-                    else if (noteType == Note.NoteType.Hopo)
-                    {
-                        if (!note.cannotBeForced && (note.isChord || !note.isNaturalHopo))
+
+                        case (Note.NoteType.Hopo):
                         {
-                            shouldBeForced = true;
+                            // Forcing consecutive same-fret HOPOs is possible in charts, but we do not allow it
+                            // (see RB2's chart of Steely Dan - Bodhisattva)
+                            if (!note.isNaturalHopo && note.cannotBeForced)
+                            {
+                                expectedForceFailure = true;
+                            }
+
+                            if (!note.cannotBeForced && (note.isChord || !note.isNaturalHopo))
+                            {
+                                shouldBeForced = true;
+                            }
+                            break;
                         }
+
+                        case (Note.NoteType.Tap):
+                        {
+                            if (!note.IsOpenNote())
+                            {
+                                note.flags |= Note.Flags.Tap;
+                                // Forced flag will be removed shortly after here
+                            }
+                            else
+                            {
+                                // Open notes cannot become taps, they become HOPOs instead
+                                expectedForceFailure = true;
```

Comment:
That's the way CH handles it, makes it so you don't have a random open strum in the middle of a tap sequence.

Reply:
Nope, Phase Shift just doesn't have open note HOPOs at all lol

Reply:
What's the Rock Band defined behaviour for this case? I would have thought that it would have naturally become a hopo open due to how close it was to the previous note.

Reply:
Oh wait, rb probably doesn't even have opens on guitar, that would explain it. CH behaviour it is.